In [1]:
%load_ext autoreload
%autoreload 2

In [2]:
import sys
import os
pathap = os.path.join('GENE_ML','gene_ml','static_sparse_grid_approximations')
sys.path.append(pathap)
from GENE_ML.gene_ml.samplers.uniform import Uniform
from GENE_ML.gene_ml.datagen.datagen import DataGen

# check config.py
from config_debug import config

parameters = ['_grp_species_0-omt','_grp_species_1-omt','species-omn']
bounds = [(4,6.7), (2.1,3.5), (0.16,2.9)] #if bounds is a float rather than tuple then the sampler will make that the only value

num_samples = 8

sampler_uniform = Uniform(parameters=parameters, num_samples=num_samples ,bounds=bounds)

# Folder name for the data to be saved in within the self.remote_save_base_dir of config
remote_save_name = "code_debug_simtimelim"

# This is used to identify the problem folder in gene
ex_id = 'code_debug_simtimelim'

single_run_timelim = 5
single_run_simtimelim = 1e-5
datagen = DataGen(config, sampler_uniform, 
                  ex_id=ex_id, remote_save_name=remote_save_name, 
                  single_run_timelim=single_run_timelim, 
                  single_run_simtimelim=single_run_simtimelim, num_workers=2)

/home/djdaniel/DEEPlasma/GENE_ML/gene_ml/runners/GENErunner.py:90: SyntaxWarning: invalid escape sequence '\d'
  sbatch_id = re.search('(?<![\d])\d{7}(?![\d])', out).group(0)
/home/djdaniel/DEEPlasma/GENE_ML/gene_ml/runners/GENErunner.py:218: SyntaxWarning: invalid escape sequence '\d'
  batch_id = re.search('(?<![\d])\d{7}(?![\d])', out).group(0)
/home/djdaniel/anaconda3/envs/gene_ml/lib/python3.12/site-packages/paramiko/pkey.py:82: CryptographyDeprecationWarning: TripleDES has been moved to cryptography.hazmat.decrepit.ciphers.algorithms.TripleDES and will be removed from this module in 48.0.0.
  "cipher": algorithms.TripleDES,
/home/djdaniel/anaconda3/envs/gene_ml/lib/python3.12/site-packages/paramiko/transport.py:219: CryptographyDeprecationWarning: Blowfish has been moved to cryptography.hazmat.decrepit.ciphers.algorithms.Blowfish and will be removed from this module in 45.0.0.
  "class": algorithms.Blowfish,
/home/djdaniel/anaconda3/envs/gene_ml/lib/python3.12/site-packages/param

In [3]:
datagen.executor.pre_run_check()

NameError: name 'datagen' is not defined

In [3]:
datagen.executor.start_runs()

EXECUTING BATCHES
PRE RUN CHECK


SINGLE RUN TIMELIM 5 N SAMPLES 4
THE ESTIMATED WALLTIME FOR RUN ex-code_debug_simtimelim_batch-0 is 00-00:00:22, dd-hh-mm-ss TO RUN 4 SAMPLES
CHECKING IF PROBLEM DIRECTORY EXISTS?
PLACING BASE PARAMETERS AND SBATCH TO REMOTE PROBLEM DIRECTORY
ALTERING THE PARAMETERS FILE IN THE REMOTE PROBLEM DIRECTORY
46 timelim 5
    timelim = 1

    timelim = 5



CODE RUN: SETTING SIMULATION TIME LIMMIT


47 simtimelim 1e-05
    simtimelim = 6.584296553193289e-11

    simtimelim = 1e-05

SBATCH
WRITE SBATCH
#!/bin/bash -l
## LUMI-C (CPU partition) submit script template
## Submit via: sbatch submit.cmd (parameters below can be overwritten by command line options)
#SBATCH -t 00-00:00:22  ## wallclock limit, dd-hh:mm:ss#SBATCH -N 1                       # total number of nodes, 2 CPUs with 64 rank each
#SBATCH --ntasks=4      # 64 per CPU (i.e. 128 per node). Additional 2 hyperthreads disabled
#SBATCH --mem=1000                    # Allocate all the memory on each no

In [4]:
datagen.executor.wait_till_finished()


CHECKING IF JOBS FINISHED: ['8107832', '8107833']
NONE OF THE INPUTED SBATCH IDs ARE RUNNING




In [6]:
import numpy as np
datagen.executor.continue_increment(group_var=['general','ntimesteps'],values = np.arange(5,20,5))


CHECKING IF JOBS FINISHED: ['8107832', '8107833']
NONE OF THE INPUTED SBATCH IDs ARE RUNNING


CONTINUING RUN ['ex-code_debug_simtimelim_batch-0', 'ex-code_debug_simtimelim_batch-1']
 with new ['general', 'ntimesteps']:5
PERFORMING STATUS CHECK
CHECK GENE STATUS: ["b'ffff'", "b'ffff'"]
ALTERING THE PROB DIR PARAMETERS FILE: /project/project_462000451/gene/auto_prob_ex-code_debug_simtimelim_batch-0/parameters
45 ntimesteps 5
    ntimesteps = 5

    ntimesteps = 5

ALTERING THE in_par DIR PARAMETERS FILES
47 ntimesteps 5
    ntimesteps = 5

    ntimesteps = 5

47 ntimesteps 5
    ntimesteps = 5

    ntimesteps = 5

47 ntimesteps 5
    ntimesteps = 5

    ntimesteps = 5

47 ntimesteps 5
    ntimesteps = 5

    ntimesteps = 5

ALTERING THE OUT DIR PARAMETERS FILE
45 ntimesteps 5
    ntimesteps = 5

    ntimesteps = 5

ALTERING THE STATUS FILE TO CONTAIN ONLY s
STATUS FOUND ffff
NEW STATUS IN FILE ssss
RUNNING continue.cmd IN THE PROBLEM DIRECTORY
CONTINUING RUN - ex-code_debug_simtimelim_

/home/djdaniel/DEEPlasma/GENE_ML/gene_ml/runners/GENErunner.py:90: SyntaxWarning: invalid escape sequence '\d'
  sbatch_id = re.search('(?<![\d])\d{7}(?![\d])', out).group(0)
/home/djdaniel/DEEPlasma/GENE_ML/gene_ml/runners/GENErunner.py:218: SyntaxWarning: invalid escape sequence '\d'
  batch_id = re.search('(?<![\d])\d{7}(?![\d])', out).group(0)


ALTERING THE OUT DIR PARAMETERS FILE
45 ntimesteps 10
    ntimesteps = 5

    ntimesteps = 10

ALTERING THE STATUS FILE TO CONTAIN ONLY s
STATUS FOUND ffff
NEW STATUS IN FILE ssss
RUNNING continue.cmd IN THE PROBLEM DIRECTORY
CONTINUING RUN - ex-code_debug_simtimelim_batch-1
OUT: Submitted batch job 8107850
 ERR: 
SUBMITTED SBATCH ID: 8107850

CHECKING IF JOBS FINISHED: ['8107832', '8107833', '8107844', '8107845', '8107848', '8107850']
FINISHED [ True  True  True  True False False]
RUNNING SBATCH IDs:  ['8107848' '8107850']


TIME SINCE STARTED  : 00-00:00:5
MAX WALL TIME       : 00-00:00:22

CHECKING IF JOBS FINISHED: ['8107832', '8107833', '8107844', '8107845', '8107848', '8107850']
FINISHED [ True  True  True  True False False]
RUNNING SBATCH IDs:  ['8107848' '8107850']


TIME SINCE STARTED  : 00-00:00:11
MAX WALL TIME       : 00-00:00:22

CHECKING IF JOBS FINISHED: ['8107832', '8107833', '8107844', '8107845', '8107848', '8107850']
FINISHED [ True  True  True  True False False]
RUNN


CHECKING IF JOBS FINISHED: ['8106219', '8106220']
NONE OF THE INPUTED SBATCH IDs ARE RUNNING


CONTINUING RUN ['ex-code_debug_simtimelim_batch-0', 'ex-code_debug_simtimelim_batch-1']
 with new ['general', 'ntimesteps']:5
PERFORMING STATUS CHECK
/home/djdaniel/DEEPlasma/GENE_ML/gene_ml/runners/GENErunner.py:90: SyntaxWarning: invalid escape sequence '\d'
  sbatch_id = re.search('(?<![\d])\d{7}(?![\d])', out).group(0)
/home/djdaniel/DEEPlasma/GENE_ML/gene_ml/runners/GENErunner.py:218: SyntaxWarning: invalid escape sequence '\d'
  batch_id = re.search('(?<![\d])\d{7}(?![\d])', out).group(0)



CHECKING IF JOBS FINISHED: ['8106563', '8106565', '8106676', '8106677']
NONE OF THE INPUTED SBATCH IDs ARE RUNNING


CONTINUING RUN ['ex-code_debug_simtimelim_batch-0', 'ex-code_debug_simtimelim_batch-1']
 with new ['general', 'ntimesteps']:5
PERFORMING STATUS CHECK


In [14]:
li = [1,2,3,4]

mask = [True, False, False , True]

import numpy as np
print(np.array(li)[mask])

[1 4]


In [12]:
import f90nml

with open('nmll', 'r') as nml_file_read:
    nml_str = str(f90nml.read(nml_file_read))
with open('nmll', 'w') as nml_file_write:
    patch= {'general':{'bpar':'nothing really matters'}}
    f90nml.patch(nml_str, patch, nml_file_write)



FileNotFoundError: [Errno 2] No such file or directory: "&parallelization\n    n_procs_s = 1\n    n_procs_v = 1\n    n_procs_w = 2\n    n_procs_x = 1\n    n_procs_y = 1\n    n_procs_z = 2\n    n_procs_sim = 4\n    n_parallel_sims = 1\n/\n\n&box\n    n_spec = 2\n    nx0 = 2\n    nky0 = 1\n    nz0 = 4\n    nv0 = 6\n    nw0 = 2\n    lv = 2\n    lw = 2\n    x0 = 2\n    kymin = 48.0\n    kx_center = 0.0\n/\n\n&in_out\n    diagdir = '/scratch/project_462000451/gene_out/gene_auto/code_debug_simtimelim/ex-code_debug_simtimelim_batch-0/scanfiles0000'\n    read_checkpoint = .false.\n    write_checkpoint = .true.\n    istep_nrg = 10000\n    istep_omega = 10000\n    istep_field = 10000\n    istep_energy = 10000\n    istep_mom = 10000\n    istep_vsp = 10000\n    istep_schpt = 10000\n/\n\n&general\n    bpar = .true.\n    nonlinear = .false.\n    x_local = .true.\n    arakawa_zv = .false.\n    comp_type = 'IV'\n    calc_dt = .true.\n    ntimesteps = 5\n    timelim = 1\n    simtimelim = 6.584296553193289e-11\n    hyp_z = -2\n    hyp_v = 0.2\n    init_cond = 'alm'\n    beta = 0.0022366753\n    debye2 = 0.00023915662\n    tau = 1.7\n    collision_op = 'landau'\n    coll = 0.0042788004\n    coll_cons_model = 'xu_rosenbluth'\n/\n\n&miscellaneous\n    dpdx_pm = -1\n    dpdx_term = 'full_drift'\n/\n\n&geometry\n    magn_geometry = 'miller'\n    trpeps = 0.25643\n    q0 = 2.2851\n    shat = 2.2142\n    amhd = -1\n    drr = -0.14686\n    drz = -0.076232\n    kappa = 1.349\n    s_kappa = 0.23129\n    delta = 0.055595\n    s_delta = 0.14461\n    zeta = -0.012412\n    s_zeta = -0.063568\n    minor_r = 1\n    major_r = 3.1481\n/\n\n&species\n    name = 'i'\n    omn = 15.265123\n    omt = 28.287016\n    mass = 1.0\n    temp = 1.0\n    dens = 1.0\n    charge = 1\n/\n\n&species\n    name = 'e'\n    omn = 15.265123\n    omt = 28.287016\n    mass = 0.0002724\n    temp = 1.0\n    dens = 1.0\n    charge = -1\n/\n\n&units\n    lref = 0.760585884664\n    mref = 2.0\n    tref = -1\n    nref = -1\n    bref = -1\n/\n\n&scan\n    scan_dims = 4, 1, 1, 1\n    par_in_dir = '/scratch/project_462000451/gene_out/gene_auto/code_debug_simtimelim/ex-code_debug_simtimelim_batch-0/scanfiles0000/in_par'\n/"

In [11]:
import f90nml
import io

# Original namelist content
original_nml_content = """
&my_namelist
    my_variable = 1 !hey hey 
/
"""

# Create a file-like object for the original namelist
original_nml_file = io.StringIO(original_nml_content)

# Read the namelist from the file-like object
nml = f90nml.read(original_nml_file)

# Modify the values
nml['my_namelist']['my_variable'] = 42

# Create a file-like object for the new namelist
new_nml_file = io.StringIO()

# Write the modified namelist to the new file-like object
nml.write(new_nml_file)

# Get the content of the new namelist
new_nml_content = new_nml_file.getvalue()

# Print the new namelist content
print(new_nml_content)


&my_namelist
    my_variable = 42
/



In [23]:
datagen.executor.check_finished()

SBATCH IDS ['8103483', '8103484']

CHECKING IF JOBS FINISHED: ['8103483', '8103484']
FINISHED [False, False]
RUNNING SBATCH IDs:  []


EXECUTOR CHECK FINISHED False


False

In [5]:
datagen.executor.wait_till_finished()

SBATCH IDS ['8103778', '8103780']

CHECKING IF JOBS FINISHED: ['8103778', '8103780']
FINISHED [False, False]
RUNNING SBATCH IDs:  []


EXECUTOR CHECK FINISHED False
TIME SINCE STARTED  : 00-00:00:6
MAX WALL TIME       : 00-00:00:22
SBATCH IDS ['8103778', '8103780']

CHECKING IF JOBS FINISHED: ['8103778', '8103780']
FINISHED [False, False]
RUNNING SBATCH IDs:  []


EXECUTOR CHECK FINISHED False
TIME SINCE STARTED  : 00-00:00:11
MAX WALL TIME       : 00-00:00:22
SBATCH IDS ['8103778', '8103780']

CHECKING IF JOBS FINISHED: ['8103778', '8103780']
FINISHED [False, False]
RUNNING SBATCH IDs:  []


EXECUTOR CHECK FINISHED False
TIME SINCE STARTED  : 00-00:00:18
MAX WALL TIME       : 00-00:00:22
SBATCH IDS ['8103778', '8103780']

CHECKING IF JOBS FINISHED: ['8103778', '8103780']
FINISHED [False, False]
RUNNING SBATCH IDs:  []


EXECUTOR CHECK FINISHED False
TIME SINCE STARTED  : 00-00:00:23
MAX WALL TIME       : 00-00:00:22
SBATCH IDS ['8103778', '8103780']

CHECKING IF JOBS FINISHED: ['81037

In [10]:
import numpy as np
single_run_simtimelim_set = np.linspace(1,2,3)
datagen.executor.increment_sim_time_lim(single_run_simtimelim_set)

### FIX ALTER PARAMETERS FILe, run it many times in the same file to watch it break it

#########!!!!!!!!!!!!FIX PARSER ALTER PARAMS FILE

set [1.  1.5 2. ]
SBATCH IDS ['8103778', '8103780']

CHECKING IF JOBS FINISHED: ['8103778', '8103780']
NONE OF THE INPUTED SBATCH IDs ARE RUNNING


EXECUTOR CHECK FINISHED True
PRAMIKO LOOKING INSIDE: /scratch/project_462000451/gene_out/gene_auto/code_debug_simtimelim/ex-code_debug_simtimelim_batch-0 
 /scratch/project_462000451/gene_out/gene_auto/code_debug_simtimelim 
 ex-code_debug_simtimelim_batch-0
SCANFILES_DIR ['scanfiles0000']
SCANFILES_NUMBER [['0000']]
LATEST_SCANFILE scanfiles0000
PRAMIKO LOOKING INSIDE: /scratch/project_462000451/gene_out/gene_auto/code_debug_simtimelim/ex-code_debug_simtimelim_batch-1 
 /scratch/project_462000451/gene_out/gene_auto/code_debug_simtimelim 
 ex-code_debug_simtimelim_batch-1
SCANFILES_DIR ['scanfiles0000']
SCANFILES_NUMBER [['0000']]
LATEST_SCANFILE scanfiles0000
using paramiko 2
using paramiko 2
CHECK GENE STATUS: ["b'ffff\\n'", "b'ffff'"]
EXECUTOR, LIST OF COMPLETE SBATCH IDs [True, True]
ALL SBATCH IDs ARE COMPLETE FOR THIS EXECUTOR
STARTIN

KeyError: 'general'

In [40]:
from GENE_ML.gene_ml.parsers.GENEparser import GENE_scan_parser
from config_debug import Config
config = Config()
parser = GENE_scan_parser(config)
# parser.alter_parameters_file('/home/djdaniel/DEEPlasma/nmll', group_var=['general','simtimelim'], value='yeahhh buddy')
parser.alter_parameters_file('/project/project_462000451/gene/auto_prob_ex-code_debug_simtimelim_batch-0/parameters_1', group_var=['general','simtimelim'], value=10.)
value = parser.get_parameter_value('/project/project_462000451/gene/auto_prob_ex-code_debug_simtimelim_batch-0/parameters_1', group_var=['general','simtimelim'])
print(value)

#!!! fix alter parameters file

47 simtimelim 10.0
    simtimelim = 50.0

    simtimelim = 10.0

gpv, paramater path /project/project_462000451/gene/auto_prob_ex-code_debug_simtimelim_batch-0/parameters_1
checking if get_p_v alters the parameters file b"&parallelization\n    n_procs_s = 1\n    n_procs_v = 1\n    n_procs_w = 2\n    n_procs_x = 1\n    n_procs_y = 1\n    n_procs_z = 2\n    n_procs_sim = 4\n    n_parallel_sims = 1\n/\n\n&box\n    n_spec = 2\n    nx0 = 2\n    nky0 = 1\n    nz0 = 4\n    nv0 = 6\n    nw0 = 2\n    lv = 2\n    lw = 2\n    x0 = 2\n    kymin = 48.0\n    kx_center = 0.0\n/\n\n&in_out\ndiagdir = '/scratch/project_462000451/gene_out/gene_auto/code_debug_simtimelim/ex-code_debug_simtimelim_batch-0/scanfiles0001' \n    read_checkpoint = .false.\n    write_checkpoint = .true.\n    istep_nrg = 10000\n    istep_omega = 10000\n    istep_field = 10000\n    istep_energy = 10000\n    istep_mom = 10000\n    istep_vsp = 10000\n    istep_schpt = 10000\n/\n\n&general\n    bpar = .true.\n    nonlinear = .false.

In [11]:
with parser.open_file('/home/djdaniel/DEEPlasma/nmll', 'r') as file:
    for line in file:
        print(line)

&parallelization

    n_procs_s = 1 !! scanned parameter

    n_procs_v = 1

    n_procs_w = 2

    n_procs_x = 1

    n_procs_y = 1

    n_procs_z = 2

    n_procs_sim = 4

    n_parallel_sims = 1

/



&box

    n_spec = 2

    nx0 = 2

    nky0 = 1

    nz0 = 4

    nv0 = 6

    nw0 = 2

    lv = 2

    lw = 2

    x0 = 2

    kymin = 48.0

    kx_center = 0.0

/



&in_out

    diagdir = '/scratch/project_462000451/gene_out/gene_auto/'

    read_checkpoint = .false.

    write_checkpoint = .true.

    istep_nrg = 10000

    istep_omega = 10000

    istep_field = 10000

    istep_energy = 10000

    istep_mom = 10000

    istep_vsp = 10000

    istep_schpt = 10000

/



&general

    bpar = .true.

    nonlinear = .false.

    x_local = .true.

    arakawa_zv = .false.

    comp_type = 'IV'

    calc_dt = .true.

    ntimesteps = 5

    timelim = 1

    simtimelim = 6.584296553193289e-11

    hyp_z = -2

    hyp_v = 0.2

    init_cond = 'alm'

    beta = 0.0022366753

    debye2 = 0

In [12]:
print(single_run_simtimelim_set)

[0.001  0.5005 1.    ]


In [7]:
's' in 'sssfs'

True

In [9]:
stdin, stdout, stderr = config.paramiko_ssh_client.exec_command('ls')
print(str(stdout.read()))

b'conda.sh\ncore\ndask_slurm_prolog_test.py\ndask_slurm_prolog_test.sh\nDesktop\nDocuments\nDownloads\nds.sif\nEasyBuild\nec.jpg\nenchanted-surrogates\nexports.sh\nfortran_lab\ngene\ngene-diag\nGENE_UncertaintyQuantification_NeuralNetworkSurrogate\nHommer2.gif\nindex.html\nMusic\nondemand\nparamiko_works\nPictures\nPublic\nsingularity_depo\nslurm-6944124.out\nslurm-6944129.out\nslurm-6944173.out\nslurm-6944579.out\nsoft\nspackTest\nstatic-sparse-grid-approximations\nstuff\nsummerschool\nTemplates\ntestsingularity\nubuntu_22.04.sif\nVideos\n'


In [9]:
%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [1]:
from GENE_ML.gene_ml.dataset.ScanData import ScanData
from config_debug import Config
config = Config()
from GENE_ML.gene_ml.parsers.GENEparser import GENE_scan_parser
import os

parser = GENE_scan_parser(config, os.path.join(config.remote_save_base_dir,'data_gen_tutorial'))


data_set = ScanData('data_gen_tutorial_latest', scan_files_target='latest', parser=parser, host=config.host, config=config, remote_path='/scratch/project_462000451/gene_out/gene_auto/data_gen_tutorial/',  test_percentage=0)


/home/djdaniel/anaconda3/envs/gene_ml/lib/python3.12/site-packages/paramiko/pkey.py:82: CryptographyDeprecationWarning: TripleDES has been moved to cryptography.hazmat.decrepit.ciphers.algorithms.TripleDES and will be removed from this module in 48.0.0.
  "cipher": algorithms.TripleDES,
/home/djdaniel/anaconda3/envs/gene_ml/lib/python3.12/site-packages/paramiko/transport.py:219: CryptographyDeprecationWarning: Blowfish has been moved to cryptography.hazmat.decrepit.ciphers.algorithms.Blowfish and will be removed from this module in 45.0.0.
  "class": algorithms.Blowfish,
/home/djdaniel/anaconda3/envs/gene_ml/lib/python3.12/site-packages/paramiko/transport.py:243: CryptographyDeprecationWarning: TripleDES has been moved to cryptography.hazmat.decrepit.ciphers.algorithms.TripleDES and will be removed from this module in 48.0.0.
  "class": algorithms.TripleDES,


Initialising dataset
SSH PATH lumi:/scratch/project_462000451/gene_out/gene_auto/data_gen_tutorial/
SCAN LOG PATH /home/djdaniel/DEEPlasma/scanlogs/data_gen_tutorial_latest

LOADING FROM SCANLOG DIR: /home/djdaniel/DEEPlasma/scanlogs/data_gen_tutorial_latest
LOG PATHS ['scan_batch-0_scanfiles-0.log' 'scan_batch-1_scanfiles-0.log']

LOADING SCANLOG AND TIME INTO PANDAS DATAFRAME /home/djdaniel/DEEPlasma/scanlogs/data_gen_tutorial_latest/scan_batch-0_scanfiles-0.log : None

LOADING SCANLOG AND TIME INTO PANDAS DATAFRAME /home/djdaniel/DEEPlasma/scanlogs/data_gen_tutorial_latest/scan_batch-1_scanfiles-0.log : None

5 SAMPLES RAN OUT OF 5 BEFORE MAX WALLTIME:
NUMBER OF SAMPLES AFTER REMOVING NaN's: 5
NaN PERCENTAGE =  0.0
SETTING VARIABLES
TEST PERCENTAGE IS 0, NO SPLIT


/home/djdaniel/DEEPlasma/GENE_ML/gene_ml/dataset/ScanData.py:209: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.
  self.df_inc_nan = pd.concat(dfs_inc_nans)
/home/djdaniel/DEEPlasma/GENE_ML/gene_ml/dataset/ScanData.py:212: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.
  return pd.concat(dfs), np.sum(n_samp_all), np.sum(n_requested_all), np.sum(n_samp_nonan_all)


In [2]:
status = parser.hit_simtimelim_test(geneerr_path='/scratch/project_462000451/gene_out/gene_auto/stability_criteria/ex-stability_criteria_batch-0/scanfiles0000/geneerr.log', get_status=True)

In [4]:
'f' in status

True

In [35]:
data_set.df

run_time      omt2      omt1      omn1 growthrate frequency
0       NaN  2.430981  6.471158  2.167879     2.6480   -8.5330
1       NaN  3.286372  5.960762  0.173329     5.2010  -42.6510
2       NaN  2.912636  4.186166  1.441729     6.3630  -21.1830
3       NaN  2.757717  5.508589  1.491274     5.6310  -19.1390
4       NaN  2.198039  6.352449  0.728789     1.6170  -24.3950

In [29]:
import re

a = int(re.findall("[0-9]{4}",'scanfiles0001')[0])
print(a)

1


In [7]:
data_set.df

run_time      omt2      omt1       omn1     shat1       q01     beta1  \
0      3.972  55.19910  33.96021  52.008700  1.315692  3.843630  0.001105   
1      1.705  54.62636  69.78608  51.447960  1.498748  5.092838  0.003788   
7      1.906  61.25873  56.25225  18.619810  1.736063  5.487717  0.001865   
10    21.402  33.22105  37.66978  12.453850  1.773974  3.949417  0.003445   
13     2.036  37.51825  39.12765  59.017660  1.083528  3.687322  0.002222   
..       ...       ...       ...        ...       ...       ...       ...   
239    7.657  66.60476  35.13247  45.245700  3.121494  5.193880  0.001799   
240   14.749  60.72701  20.90660  40.640050  1.429262  5.057919  0.001787   
241   11.208  30.88569  16.21406   6.317262  2.238682  5.506229  0.002725   
244   15.154  27.77436  26.30215  20.327830  3.668582  4.951364  0.003851   
248    1.546  60.39266  19.62775  54.221810  2.759035  5.126551  0.003441   

       kymin1 growthrate frequency  
0    0.225655     5.3950    9.1130  
1    0.239661    10.1140   13.8750  
7    0.115503     9.8640    4.0210  
10   0.475377     1.9310   -2.5220  
13   0.108494    11.8230    5.1930  
..        ...        ...       ...  
239  0.298063     3.5080   10.9830  
240  0.364511     3.3100   10.1990  
241  0.475426     3.1610    4.3160  
244  0.856212     0.5890   -8.3100  
248  0.154241    13.3870    4.7470  

[2508 rows x 10 columns]

In [9]:
from GENE_ML.gene_ml.dataset.ScanData import ScanData
from config_debug import config

data_set = ScanData('code_debug_tester', parser=datagen.parser, host=config.host, remote_path=datagen.remote_save_dir,  test_percentage=50)


<class 'paramiko.sftp_file.SFTPFile'>
Initialising dataset
SSH PATH lumi:/scratch/project_462000451/gene_out/gene_auto/code_debug_tester
SCAN LOG PATH /home/djdaniel/DEEPlasma/scanlogs/code_debug_tester

LOADING FROM SCANLOG DIR: /home/djdaniel/DEEPlasma/scanlogs/code_debug_tester
LOG PATHS ['geneerr_batch-0_scanfiles-0.log' 'geneerr_batch-1_scanfiles-0.log'
 'scan_batch-0_scanfiles-0.log' 'scan_batch-1_scanfiles-0.log']

LOADING SCANLOG AND TIME INTO PANDAS DATAFRAME /home/djdaniel/DEEPlasma/scanlogs/code_debug_tester/scan_batch-0_scanfiles-0.log : /home/djdaniel/DEEPlasma/scanlogs/code_debug_tester/geneerr_batch-0_scanfiles-0.log

LOADING SCANLOG AND TIME INTO PANDAS DATAFRAME /home/djdaniel/DEEPlasma/scanlogs/code_debug_tester/scan_batch-1_scanfiles-0.log : /home/djdaniel/DEEPlasma/scanlogs/code_debug_tester/geneerr_batch-1_scanfiles-0.log

0 SAMPLES RAN OUT OF 8 BEFORE MAX WALLTIME:
NUMBER OF SAMPLES AFTER REMOVING NaN's: 0
NaN PERCENTAGE =  nan
SETTING VARIABLES

RANDOMLY SPLITTIN

/home/djdaniel/DEEPlasma/GENE_ML/gene_ml/dataset/ScanData.py:61: RuntimeWarning: invalid value encountered in scalar divide
  nan_percentage = (n_samp-n_samp_nonan)*100/n_samp


ValueError: With n_samples=0, test_size=0.5 and train_size=None, the resulting train set will be empty. Adjust any of the aforementioned parameters.

In [19]:
data_set.df

run_time      omt2      omt1      omn1 growthrate frequency
0   28.576  3.381341  4.499231  1.386877     8.5470  -25.4860
1   31.241  3.116692  4.539037  2.544643     5.6990   -9.4170
2  107.051  2.196530  6.083308  2.124313     1.2290   -7.2660
3   44.102  2.882231  5.836209  0.328710     3.6090  -36.6460
4   24.030  3.319788  5.275577  1.054928     7.7850  -29.4530

In [20]:
data_set.x

array([[3.381341 , 4.499231 , 1.386877 ],
       [3.116692 , 4.539037 , 2.544643 ],
       [2.19653  , 6.083308 , 2.124313 ],
       [2.882231 , 5.836209 , 0.3287098],
       [3.319788 , 5.275577 , 1.054928 ]])

In [25]:
data_set.x_train, data_set.growthrate_train

(array([[3.319788, 5.275577, 1.054928],
        [3.381341, 4.499231, 1.386877]]),
 array([7.785, 8.547]))

In [24]:
data_set.x_test, data_set.growthrate_test

(array([[2.19653  , 6.083308 , 2.124313 ],
        [3.116692 , 4.539037 , 2.544643 ],
        [2.882231 , 5.836209 , 0.3287098]]),
 array([1.229, 5.699, 3.609]))

# Loading data already saved in scanlogs


In [27]:
from GENE_ML.gene_ml.parsers.GENEparser import GENE_scan_parser
from config import config

parser = GENE_scan_parser(config.save_dir, config.base_params_path)

data_set = ScanData('data_gen_tutorial4', parser=parser, host=None, remote_path=None,  test_percentage=50)


Initialising dataset
SSH PATH None:None
SCAN LOG PATH /home/djdaniel/DEEPlasma/scanlogs/data_gen_tutorial4

LOADING FROM SCANLOG DIR: /home/djdaniel/DEEPlasma/scanlogs/data_gen_tutorial4
LOG PATHS ['geneerr_batch-0_scanfiles-0.log' 'scan_batch-0_scanfiles-0.log']

LOADING SCANLOG AND TIME INTO PANDAS DATAFRAME /home/djdaniel/DEEPlasma/scanlogs/data_gen_tutorial4/scan_batch-0_scanfiles-0.log : /home/djdaniel/DEEPlasma/scanlogs/data_gen_tutorial4/geneerr_batch-0_scanfiles-0.log

5 SAMPLES RAN OUT OF 5 BEFORE MAX WALLTIME:
NUMBER OF SAMPLES AFTER REMOVING NaN's: 5
NaN PERCENTAGE =  0.0
SETTING VARIABLES

RANDOMLY SPLITTING DATA INTO TEST AND TRAINING SETS: 50% test, 50 training.


# Combine datasets

In [29]:
data_set_1 = ScanData('data_gen_tutorial4', parser=parser, host=None, remote_path=None,  test_percentage=50)
data_set_2 = ScanData('data_gen_tutorial4', parser=parser, host=None, remote_path=None,  test_percentage=50)

data_set_3 = data_set_1.concat([data_set_2])

data_set_3.df

Initialising dataset
SSH PATH None:None
SCAN LOG PATH /home/djdaniel/DEEPlasma/scanlogs/data_gen_tutorial4

LOADING FROM SCANLOG DIR: /home/djdaniel/DEEPlasma/scanlogs/data_gen_tutorial4
LOG PATHS ['geneerr_batch-0_scanfiles-0.log' 'scan_batch-0_scanfiles-0.log']

LOADING SCANLOG AND TIME INTO PANDAS DATAFRAME /home/djdaniel/DEEPlasma/scanlogs/data_gen_tutorial4/scan_batch-0_scanfiles-0.log : /home/djdaniel/DEEPlasma/scanlogs/data_gen_tutorial4/geneerr_batch-0_scanfiles-0.log

5 SAMPLES RAN OUT OF 5 BEFORE MAX WALLTIME:
NUMBER OF SAMPLES AFTER REMOVING NaN's: 5
NaN PERCENTAGE =  0.0
SETTING VARIABLES

RANDOMLY SPLITTING DATA INTO TEST AND TRAINING SETS: 50% test, 50 training.
Initialising dataset
SSH PATH None:None
SCAN LOG PATH /home/djdaniel/DEEPlasma/scanlogs/data_gen_tutorial4

LOADING FROM SCANLOG DIR: /home/djdaniel/DEEPlasma/scanlogs/data_gen_tutorial4
LOG PATHS ['geneerr_batch-0_scanfiles-0.log' 'scan_batch-0_scanfiles-0.log']

LOADING SCANLOG AND TIME INTO PANDAS DATAFRAME /ho

(  run_time      omt2      omt1      omn1 growthrate frequency
 0   28.576  3.381341  4.499231  1.386877     8.5470  -25.4860
 1   31.241  3.116692  4.539037  2.544643     5.6990   -9.4170
 2  107.051  2.196530  6.083308  2.124313     1.2290   -7.2660
 3   44.102  2.882231  5.836209  0.328710     3.6090  -36.6460
 4   24.030  3.319788  5.275577  1.054928     7.7850  -29.4530
 0   28.576  3.381341  4.499231  1.386877     8.5470  -25.4860
 1   31.241  3.116692  4.539037  2.544643     5.6990   -9.4170
 2  107.051  2.196530  6.083308  2.124313     1.2290   -7.2660
 3   44.102  2.882231  5.836209  0.328710     3.6090  -36.6460
 4   24.030  3.319788  5.275577  1.054928     7.7850  -29.4530,
 array([[2.882231 , 5.836209 , 0.3287098],
        [3.381341 , 4.499231 , 1.386877 ],
        [3.319788 , 5.275577 , 1.054928 ],
        [3.116692 , 4.539037 , 2.544643 ],
        [2.19653  , 6.083308 , 2.124313 ]]))